resin/auth.py

```diff
@@ -105,9 +105,13 @@ def is_logged_in(self):
 
         """
 
-        return self.base_request.request(
-            '/whoami', 'GET', endpoint=self.settings.get('api_endpoint')
-        )
+        try:
+            self.base_request.request(
+                '/whoami', 'GET', endpoint=self.settings.get('api_endpoint')
+            )
+            return True
+        except:
```

Comment:
@dfunckt this method will check if the user is logged or not (haven't logged in, expired token, invalid api key are considered as not logged in), so basically if that API call doesn't raise any error, it means user is logged in. That's why a general catch is fine here.